backend/src/services/round.service.ts

```diff
@@ -0,0 +1,121 @@
+import { Game, Player, SetupType } from "../models";
+import { getGame } from "./game.service";
+import { RoundState } from "../models/round.model";
+import { ErrorType, ServiceError } from "../util";
+
+export const enterPlayersChooseState = async (
+  gameCode: Game["gameCode"],
+  playerId: Player["id"]
+): Promise<void> => {
+  const game = await getGame(gameCode);
+  const round = game.rounds.slice(-1)[0];
+
+  if (
+    round !== undefined &&
+    round.state === RoundState.before &&
+    round.host.toString() === playerId
+  ) {
+    round.state = RoundState.playersChoose;
+
+    // TODO: Allocate player cards here?
```

Comment:
Do you mean players drawing a new card here?

Reply:
Yeah although I think it is best to have the allocation function in the game service, it's just a better of where and when that function is called from & it has multiple answers